From bc8c2f309c9e73e114ee28284ede05a5d22caafc Mon Sep 17 00:00:00 2001
From: Mateusz Guzik <mjguzik@gmail.com>
Date: Thu, 13 Aug 2026 19:41:39 +0800
Subject: [PATCH 1/2] fs: mark lookup_slow() as noinline

Keep lookup_slow() out of walk_component() so the compiler does not reserve extra registers for the uncommon slow lookup path. This shortens and simplifies the path-walk fast path.

Backport of upstream commit 8d79ec9e7f634e10c6cdc7f3999023bd988df1ad.

Signed-off-by: Dong Chenchen <superdcc97@163.com>
---
 fs/namei.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/fs/namei.c b/fs/namei.c
index 6a95095447dc..d2de0daef76b 100644
--- a/fs/namei.c
+++ b/fs/namei.c
@@ -1773,9 +1773,9 @@ static struct dentry *__lookup_slow(const struct qstr *name,
 	return dentry;
 }
 
-static struct dentry *lookup_slow(const struct qstr *name,
-				  struct dentry *dir,
-				  unsigned int flags)
+static noinline struct dentry *lookup_slow(const struct qstr *name,
+					   struct dentry *dir,
+					   unsigned int flags)
 {
 	struct inode *inode = dir->d_inode;
 	struct dentry *res;
-- 
2.43.0

